### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by excitedsoup**

### CODE REVIEW

File: hw01-tolstoy.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+def main():
+    f = open('WarAndPeace.txt', 'r', encoding="utf8")
```
Comment (enkeboll) [inline hw01-tolstoy.py:2]: would prefer that this file not be included in the PR and instead be downloaded by the script

File: hw01-tolstoy.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+def main():
+    f = open('WarAndPeace.txt', 'r', encoding="utf8")
+    text = f.read()
+    f.close()
+    
+    punct = '!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~—“”'
+    
+    for i in text:
```
Comment (enkeboll) [inline hw01-tolstoy.py:8]: nice understanding of strings in python.  but it's bad practice to use `i` for non-counting integers.  would prefer something more descriptive like `symbol`

File: hw01-tolstoy.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    
+    punct = '!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~—“”'
+    
+    for i in text:
+        if i in punct:
+            text = text.replace(i, ' ')
+    
+    splitText = text.split()
+    
+    for i in range(len(splitText)):
```
Comment (enkeboll) [inline hw01-tolstoy.py:14]: this can be simplified as `for word in splitText`

File: hw01-tolstoy.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        splitText[i] = splitText[i].lower()
```
Comment (enkeboll) [inline hw01-tolstoy.py:15]: and this can be simplified as `splitText = [w.lower() for w in splitText]`

File: hw01-tolstoy.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+def main():
+    f = open('WarAndPeace.txt', 'r', encoding="utf8")
+    text = f.read()
+    f.close()
+    
+    punct = '!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~—“”'
+    
+    for i in text:
+        if i in punct:
+            text = text.replace(i, ' ')
```
Comment (enkeboll) [inline hw01-tolstoy.py:10]: this is an **incredibly** slow loop, and probably doesn't do exactly what you want it to be doing

File: hw01-tolstoy.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    
+    splitText = text.split()
+    
+    for i in range(len(splitText)):
+        splitText[i] = splitText[i].lower()
+    
+    uniqueWords = set(splitText)
+    
+    print('There are ' + str(len(uniqueWords)) + ' unique words in the provided text file of War and Peace')
+main()
```
Comment (enkeboll) [inline hw01-tolstoy.py:20]: this should be wrapped in an `if __name__ == '__main__':`

File: hw01-tolstoy.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+            text = text.replace(i, ' ')
+    
+    splitText = text.split()
+    
+    for i in range(len(splitText)):
+        splitText[i] = splitText[i].lower()
+    
+    uniqueWords = set(splitText)
+    
+    print('There are ' + str(len(uniqueWords)) + ' unique words in the provided text file of War and Peace')
```
Comment (enkeboll) [inline hw01-tolstoy.py:19]: your solution gets 19k words, which is a little low I think

**Review by enkeboll — APPROVED** [review]
Good implementation, but losing a point for following directions-- should be in directory `homework/`
9/10

**Comment by excitedsoup** [comment]
Yeah, you're right - I partially pulled the implementation from a previous
project, since I had to strip punctuation before (which had a whole
headache involving a different variation of double quote characters not
included in string.punctuation), and I'm not entirely sure why I made that
loop that way. Porting it to what I'd used in that previous project looks
like:

text = text.translate(str.maketrans('','', punct))

which makes its runtime less than a second and gives a result of 21015
words. I didn't exactly follow what the " if __name__ == '__main__': " was
talking about - where should that be and what does that do?

I appreciate the in-depth feedback.

Thanks,
Peter

**Comment by enkeboll** [comment]
RE: `if __name__ == '__main__'`, read this answer which explains it better than I can:
https://stackoverflow.com/questions/419163/what-does-if-name-main-do

tl;dr: it's a good idea to do like you did, and encapsulate your code in a function first.  It's then **great** to wrap that function call in an if statement that ensures you only call that function exactly when you intend to.  It's just good a practice to get in the habit of doing!
